packages/material-ui-unstyled/src/FormControlUnstyled/FormControlUnstyled.test.tsx

```diff
@@ -21,4 +45,165 @@ describe('<FormControlUnstyled />', () => {
       },
     },
   }));
+
+  describe('initial state', () => {
+    it('has undefined value', () => {
+      const readContext = spy();
+      render(
+        <FormControlUnstyled>
+          <TestComponent contextCallback={readContext} />
+        </FormControlUnstyled>,
+      );
+
+      expect(readContext.args[0][0]).to.have.property('value', undefined);
+      expect(readContext.args[0][0]).to.have.property('defaultValue', undefined);
```

Comment:
I think in this test you can use `expect(getByRole('textbox')).to.have.property('value', "")`. Can you go over the tests again and try not to use `readContext` and instead assert on the generated DOM? 

Right now you're creating a test component for two testing APIs but only make use of one. I'm trying to avoid test helpers local to a test as much as possible since hurts readability of a test. 

The implementation with `useEffect` is also slightly broken in React 18 but we have a workaround for that. But if we can stick with existing helpers, let's do it.